### DISCUSSION TITLE
Way to revoke/revert published lyrics and Access to submission history.

### DISCUSSION THREAD

**Discussion by Dr-Blank** [tranxuanthang/lrcget#24] [Ideas]
I am not sure how this would work but the use case is as follows.

I submitted a lyric for a song, but it turns out it was not for that track but for some other track. To correct the error, I'd like to reinstate the previous version/state of lyrics. If there were no lyrics, mark as not found since submitting blank lyrics marks it as instrumental (talking about `id: 3916049`.

That is another point of concern here, the proof of work is not that much work and a malicious party could just go on submit junk lyrics and without some sort of check/history for actual users using this service, this would be bad.

**Comment by tranxuanthang** [comment]
It's not too hard because all the previous versions are kept, it's just that there is no API for seeing that for now. I'll work on it as soon as I have free time.

I'm still thinking about requiring users creating accounts, but I feel it is too inconvenience and troublesome for just a lyrics database, and it will still face the same problems about spamming. Registration with captcha is a thing, but effective captchas from big techs are not FOSS and not self-host friendly, and have privacy tradeoff (recaptcha, hcaptcha,...).

POW can discourage abusers because they need more computing power (also mean more electricity and processing time) to spam the system. Also, the target can be smaller (meaning harder difficulty) if it sees a surge in lyrics submission in a short time frame.
